src/rendering/MarkerManager.cc

```diff
@@ -601,3 +613,11 @@ void MarkerManagerPrivate::OnMarkerMsg(const ignition::msgs::Marker &_req)
   std::lock_guard<std::mutex> lock(this->mutex);
   this->markerMsgs.push_back(_req);
 }
+
+/////////////////////////////////////////////////
+void MarkerManagerPrivate::OnMarkerMsgArray(const ignition::msgs::Marker_V &_req)
```

Comment:
does changing the return type to `bool` help with the misleading service call time out msg?

Reply:
Good catch, I'll try this out shortly

Reply:
@caguero do you know if the return type is the reason for the service call time out?

The cmd @JShep1 tested below got no response after changing to `bool`:

```
ign service -s /marker_array --reqtype ignition.msgs.Marker_V --reptype ignition.msgs.Boolean --timeout 1000 --req 'marker: {action: ADD_MODIFY, type: SPHERE, id: 4, pose: {position: {x:2.0, y:2.0, z:2.0}, orientation: {x:0.0, y:0.0, z:0.0, w:1.0}}, parent: "cylinder"}'
```

Reply:
I think I know why. You'll need to use the version of the callback with `msgs::Boolean` response arg, see [this example](https://github.com/ignitionrobotics/ign-gazebo/blob/master/src/gui/plugins/scene3d/Scene3D.cc#L2208)

Reply:
That worked for me!  Updates made [here](29ed047e36833e766e269d0203cd989144840a7c)